### ISSUE TITLE
support content based connection

### ISSUE THREAD

**Issue by lafirest** [emqx/esockd#148]
The current esockd is based on socket peerinfo to create a connection, but the survival time of the udp session on the router is generally very short. 
Coupled with the influence of NAT, the peerinfo of the same client may be constantly changing. 
So may we can support create a connection based on the content carried in the udp package

![connection_on_data](https://user-images.githubusercontent.com/22971499/128478375-0b7d1519-63e2-4d15-b993-4901c00bae35.png)

**Comment by robertzhouxh** [comment]
+1

**Comment by HJianBo** [comment]
Yep, we are planning the optimization for some UDP-based protocols, e.g CoAP, MQTT-SN, LwM2M, etc.
